backend/app/anti_recommendation_engine/anti_recommendation_engine.py

```diff
@@ -1,53 +1,72 @@
-from app.anti_recommenders.anti_recommender_generator import AntiRecommendationGenerator
-from app.models.record.record import Record
+from collections.abc import Generator
+
+from app.anti_recommenders.anti_recommender import AntiRecommender
+from app.anti_recommenders.open_ai.open_ai_normal_anti_recommender import (
+    OpenAiNormalAntiRecommender,
+)
+from app.models.anti_recommendation import AntiRecommendation
+from app.models.record import Record
+from app.models.settings import settings
 from app.readers.all_source_reader import AllSourceReader
 
 
-class AntiRecommendationEngine:
-    """
-    The main AntiRecommendationEngine for the NerdSwipe backend.
-    It retrieves AntiRecommendations from the AntiRecommendationGenerator and possesses methods
-    which return tuples of Records that have the same titles as the generated AntiRecommendations or previous AntiRecommendations.
+class AntiRecommendationEngine(AntiRecommender):
```

Comment:
This needs a better docstring to indicate that it's keeping state and delegating "new" work to an underlying `AntiRecommender`.

Reply:
I have improved the docstring in `AntiRecommendationEngine` and other classes/functions.